engine/classes/Elgg/Database/EntityTable.php

```diff
@@ -390,7 +390,7 @@ public function get($guid, $type = null, $subtype = null) {
 		$guid = (int) $guid;
 
 		$entity = $this->getFromCache($guid);
-		if ($entity && elgg_instanceof($entity, $type, $subtype)) {
+		if ($entity instanceof \ElggEntity && $entity->getType() === $type && $entity->getSubtype() === $subtype) {
```

Comment:
this line is causing the erros... need to check type/subtype differently

also add integration tests as only a weird simpletest caught this